src/app/(full-page)/moviesPage/page.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from "react";
+import { GetServerSideProps } from "next";
+import { Movie } from "../../../lib/types/movie";
+import { MovieCard } from "../../../components/movie-card";
+import { PageButton } from "../../../components/page-button";
+import axios from "axios";
+
+interface MoviesProps {
+  movies: Movie[];
+}
+
+const Movies = ({ movies }: MoviesProps) => {
+  const [currentPage, setCurrentPage] = useState(1);
```

Comment:
You can manage the pages through a query parameter in the url
(e.g. /movies?page=1)
This allows you to keep the page as a server component
The button would then be a links, that navigate to the movies page with the query paramter